kendzi3d-josm-core/src/main/java/kendzi/josm/kendzi3d/util/ModelUtil.java

```diff
@@ -64,32 +64,29 @@ public static Double parseHeight(String pHeightStr, Double pDefault) {
             return pDefault;
         }
 
-        String heightStr = pHeightStr.trim().toLowerCase();
-        heightStr = heightStr.replaceAll(",", ".");
+        pHeightStr = pHeightStr.trim().toLowerCase().replaceAll(",", ".");
 
         try {
-            if (heightStr.endsWith("m")) {
-                heightStr = heightStr.substring(0, heightStr.length() - 1);
-                return new Double(" " + heightStr + " ");
-
-            } else if (heightStr.endsWith("ft")) {
-
-                // 1 foot = 0.3048 meters
-                heightStr = heightStr.substring(0, heightStr.length() - 1);
-                return 0.3048d * new Double(" " + heightStr + " ");
-
-            } else if (heightStr.endsWith("feet")) {
-
-                // 1 foot = 0.3048 meters
-                heightStr = heightStr.substring(0, heightStr.length() - 4);
-                return 0.3048d * new Double(" " + heightStr + " ");
+            // default we take in [m]
+            double mult = 1d;
+            int strlen = pHeightStr.length();
+
+            switch (pHeightStr.replaceAll("[^a-z]", "")) {
+            case "km": mult = 1000d;     strlen--; strlen--; break;
+            case "dm": mult = .1d;       strlen--; strlen--; break;
+            case "cm": mult = .01d;      strlen--; strlen--; break;
+            case "mm": mult = .001d;     strlen--; strlen--; break;
+            case "m":                              strlen--; break;
+            case "feet":                 strlen--; strlen--;
+            case "ft": mult = 0.3048d;   strlen--; strlen--; break;
+            case "inch":                 strlen--; strlen--;
+            case "in": mult = 1/0.0254d; strlen--; strlen--; break;
```

Comment:
1 inch is 0.0254m so multipler should be "0.0254" not "1/0.0254"?

Reply:
yes, reciprocate would tells us how many inches there are in a metre
(which means 1/39.37 would have been close, but 1/0.0254 surely was not;  good eyes!)